docs/Privacy.md

```diff
@@ -24,12 +24,23 @@ short links). All this makes perfect sense and would probably be done anyway, as
 We anonymise IP addresses so there's no exact mapping back to an individual using an IP. Not that it's trivial to map an
 IP to a user anyway.
 
-We shouldn't store data forever: our web logs are set to delete after a few months.
+We shouldn't store data forever: our web logs are set to delete after 32 days. Amazon infrastructure logs
+(separate from our web logs) also contain full IP addresses and are kept for 32 days. Compilation analytics logs
+are kept for up to 1 year - these contain hashed source code (not reversible), compiler options, and usage patterns
+to help us improve the service. Lambda and API Gateway logs are kept for 7-14 days.
 
 Short URLs do turn up in the web logs: from the short URL of course one can easily extract the source code embedded in
 that short URL. Users are notified of this in the privacy policy. The ultimate recourse for users concerned about this
 is to not use the main Compiler Explorer but instead run their own local service, which is relatively straightforward.
 
+We also integrate with third-party services with user consent: Claude Explain sends code to Anthropic for analysis, and
+we use Sentry for error reporting (which keeps IP and browser info for up to 90 days - note this is controlled by Sentry's
+retention settings, not our code). Users can control whether their code is stored for diagnostic purposes through a
+setting that defaults to enabled but can be disabled.
+
+Important: The actual retention periods are configured in our terraform infrastructure, not in the application code.
```

Comment:
Is this something the end user needs to know?

Reply:
No. But they don't read this doc, this doc is for us. The privacy policy is in the pug file

Reply:
Oh right! Didn't notice which file I was in 😄